#### Artificial data generated from the F distribution ####
> Permutation values are obtained by randomly drawing samples from the F distribution with parameters (degrees of freedom) 5 and 10.  This dataset contains 1000 permutation values for each of five events/experiments.  The correct theoretical permutation test P-values for these five events are found when evaluating the cdf of the F distribution at the values of the original statistic (on the first row in the dataset).  These P-values are: 10-1, 10-2, 10-3, 10-4 and 10-5.<br>
<blockquote><a href='http://epept.googlecode.com/files/art_f.tsv'>Artificial Data Distribution example</a></blockquote>

The "Mode" for this dataset should be set to "Permutation Values".<br>
<br>
<br>
<h4>Artificial data generated from the normal distribution</h4>
<blockquote>Permutation values are obtained by randomly drawing samples from the normal distribution with zero-mean and unit-variance.  This dataset contains 10000 permutation values for each of ten events/experiments.  The correct theoretical permutation test P-values for these ten events are found when evaluating the cdf of the normal distribution at the values of the original statistic (on the first row in the dataset).  These P-values are:10-1, 10-2, 10-3, 10-4, 10-5, 10-6, 10-7, 10-8, 10-9 and 10-10.<br><br>
<a href='http://epept.googlecode.com/files/art_n.tsv'>Artificial Data Normal example</a></blockquote>

The "Mode" for this dataset should be set to "Permutation Values".<br>
<br>
<br>
<h4>Yeast gene expression data</h4>
<blockquote>The employed yeast data is the one mentioned in Section 3.2 of the <a href='http://bioinformatics.oxfordjournals.org/cgi/content/abstract/25/12/i161?eto'>paper</a>.  It contains 63 genes across 170 microarrays. In 80 of these arrays yeast was grown aerobically; for the other 90 arrays yeast was grown anaerobically; this division constitutes the labels.  Permutation values are obtained by computing the SAM statistic on the yeast expression data using permuted label configurations; the original statistic is computed by using the original label configuration.  The original dataset can be downloaded <a href='http://epept.googlecode.com/files/yeast_orgdata.tsv'>here</a>.<br><br>
This file also contains the correct permutation test P-values that are computed as explained in Section 3.2 of the paper. For each gene 10000 permutation values were generated.<br>
</blockquote><ul><li><a href='http://epept.googlecode.com/files/yeast.tsv'>Yeast Data example</a>
</li><li><a href='http://epept.googlecode.com/files/yeast.xls'>Excel file format</a></li></ul>

The "Mode" for this dataset should be set to "Permutation Values".<br>
<br>
<br>
<h4>SAM/GSEA for Response Type 'Two class unpaired'</h4>
<ul><li><a href='http://epept.googlecode.com/files/yeastdata.txt'>SAM/GSEA Data</a>
</li><li><a href='http://epept.googlecode.com/files/yeastdata_small.txt'>Smaller SAM/GSEA Data</a></li></ul>

The "Mode" for this dataset should be set to "SAM" or "GSEA".<br>
<br>
<br>
<h4>GSEA gene set file (in .gmt format)</h4>
<ul><li><a href='http://epept.googlecode.com/files/geneset.gmt'>geneset.gmt</a>
</li><li><a href='http://epept.googlecode.com/files/genesets_small.gmt'>genesets_small.gmt</a>